### DISCUSSION TITLE
Regression in v7.0.0 : AndroidOptions Color is not working anymore

### DISCUSSION THREAD

**Discussion by tranb3r** [thudugala/Plugin.LocalNotification#178] [Q&A] [answered]
In Plugin.LocalNotification.Platform.Droid.NotificationServiceImpl, I think the code for setting the color is not correct:
```
            if (string.IsNullOrWhiteSpace(request.Android.Color) == false)
            {
                var colorId = Application.Context.Resources?.GetIdentifier(request.Android.Color, "color", Application.Context.PackageName) ?? 0;
                builder.SetColor(colorId);
            }
```
SetColor expects an ARGB int, not a resourceId.
This was working in previous versions so this is a regression in v7.0.0.

Also, why did you do this change ? In the previous version, we were able to set the color value (ARGB int) directly in NotificationRequest, but now if I understand correctly you expect us to use an android resource for it ?

**Comment by thudugala** [comment]
@tranb3r Yes, the expectation is to use a colour in android resource, if it's a custom one, developer need to add it to android resource then use the name of the colour. 

This makes the code more readable.

**Comment by tranb3r** [comment]
@thudugala 
I don't understand how this makes the code more readable, since it means duplicating the color definition that is already done in the cross-platform library (for a xamarin forms application).
Also, why did you remove the bug label in this issue ? This code does not work. SetColor expects an ARGB color, not a resourceId. I've tested the sample app, it does not work anymore (the color is not applied).

**Comment by thudugala** [comment]
@tranb3r Have not updated the sample with version 7 yet.

in your experience, is it much more likely to use a custom colour, rather than existing colour definition already in android resources ?

Not a bug, as this was a design decision in version 7.0.0

**Comment by tranb3r** [comment]
@thudugala 
The sample was actually updated in commit 59bd23f.
But it does not work, since - I have to insist - there is a bug in 7.0.0 : you pass the resource id, not the resource value, to SetColor, in NotificationServiceImpl.

Also, about your design decision:
In my experience, in a xamarin forms application, colors are defined in app resources, not in android resources.
Maybe a better decision could have been to have both options:
- either set an ARGB value (like in 6.x)
- or use an android resource.

**Reply by mos379** [reply]
I agree with the fact that the colors are defined in the app resources as this is where the different schemes especially for dark and light mode are maintained.

**Comment by thudugala** [comment]
@tranb3r and @mos379 will fix it in next version that will come soon

**Comment by thudugala** [comment, marked as answer]
fixed issue after version 8.0.1.

Now you can set either an ARGB value or use an android resource name.
